<launch>
  <arg name="modified_goal_topic_name" default="/planning/scenario_planning/modified_goal"/>
  <arg name="map_topic_name" default="/map/vector_map"/>
  <arg name="route_topic_name" default="/planning/route"/>
  <arg name="route_state_topic_name" default="/planning/route_state"/>
  <arg name="clear_route_service_name" default="/planning/clear_route"/>
  <arg name="set_lanelet_route_service_name" default="/planning/set_lanelet_route"/>
  <arg name="set_waypoint_route_service_name" default="/planning/set_waypoint_route"/>
  <arg name="visualization_topic_name" default="/planning/mission_planning/route_marker"/>
  <arg name="mission_planner_param_path" default="$(find-pkg-share autoware_mission_planner_universe)/config/mission_planner.param.yaml"/>

  <node_container pkg="rclcpp_components" exec="component_container_mt" name="mission_planner_container" namespace="">
    <composable_node pkg="autoware_manual_lane_change_handler" plugin="autoware::manual_lane_change_handler::ManualLaneChangeHandler" name="manual_lane_change_handler" namespace="">
      <param from="$(var mission_planner_param_path)"/>
      <remap from="~/input/modified_goal" to="$(var modified_goal_topic_name)"/>
      <remap from="~/input/vector_map" to="$(var map_topic_name)"/>
      <remap from="~/input/odometry" to="/localization/kinematic_state"/>
      <remap from="~/input/operation_mode_state" to="/system/operation_mode/state"/>
      <remap from="~/input/reroute_availability" to="/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/output/is_reroute_available"/>
      <remap from="~/route" to="route"/>
      <remap from="~/state" to="state"/>
      <remap from="~/debug/route_marker" to="$(var visualization_topic_name)"/>
    </composable_node>
  </node_container>
</launch>
